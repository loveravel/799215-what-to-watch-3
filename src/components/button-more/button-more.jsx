import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {getFilmsCount, getFilteredFilms} from "../../reducer/data/selectors.js";
import {ActionCreator} from "../../reducer/data/data.js";

const ButtonMore = (props) => {
  const {filteredFilms, filmsCount, incrementFilmsCount} = props;

  if (filteredFilms.length <= filmsCount) {
    return null;
  }

  return (
    <button
      className="catalog__button"
      type="button"
      onClick={() => {
        incrementFilmsCount();
      }}
    >
      Show more
    </button>
  );
};

ButtonMore.propTypes = {
  filteredFilms: PropTypes.array,
  filmsCount: PropTypes.number,
  incrementFilmsCount: PropTypes.func,
};

const mapStateToProps = (state) => ({
  filmsCount: getFilmsCount(state),
  filteredFilms: getFilteredFilms(state),
});

const mapDispatchToProps = (dispatch) => ({
  incrementFilmsCount() {
    dispatch(ActionCreator.incrementFilmsCount());
  }
});

export {ButtonMore};
export default connect(mapStateToProps, mapDispatchToProps)(ButtonMore);
