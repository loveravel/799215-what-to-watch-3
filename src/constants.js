export const DEFAULT_GENRE = `All genres`;

export const EMAIL_REGEX = RegExp(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);

export const BASE_URL = `https://htmlacademy-react-3.appspot.com`;

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  MOVIE: `/films/:id`,
  ADD_REVIEW: `/films/:id/review`,
  VIDEO_PLAYER: `/player/:id`,
};

export const Tab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const FilmsCount = {
  STEP: 8,
  INITIAL_MAIN: 8,
  INITIAL_SIMILAR: 4,
};

export const RatingDescription = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very Good`,
  AWESOME: `Awesome`,
};

export const ErrorText = {
  EMAIL: `Please enter a valid email address.`,
  PASSWORD: `Please enter a valid password.`
};

export const InputName = {
  EMAIL: `email`,
  PASSWORD: `password`,
};

export const FavoriteStatus = {
  YES: 1,
  NO: 0,
};

export const CommentLength = {
  MIN: 50,
  MAX: 400,
};
