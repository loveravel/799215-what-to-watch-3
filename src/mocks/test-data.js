export const films = [
  {
    id: 1,
    name: `Movie 1`,
    genre: ``,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideo: ``,
    video: ``,
    isFavorite: true
  },
  {
    id: 2,
    name: `Movie 2`,
    genre: ``,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideoLink: ``,
    video: ``,
    isFavorite: false
  },
  {
    id: 3,
    name: `Movie 3`,
    genre: `Drama`,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideo: ``,
    video: ``,
    isFavorite: false
  },
];

export const movie = {
  id: 1,
  name: `Movie 1`,
  genre: ``,
  released: 2019,
  backgroundImage: ``,
  backgroundColor: ``,
  previewImage: ``,
  posterImage: ``,
  description: ``,
  rating: 4.5,
  scoresCount: 300,
  runTime: 3,
  director: ``,
  starring: [],
  previewVideo: ``,
  video: ``,
  isFavorite: false
};

export const reviews = [
  {
    id: 1,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  }
];

export const userData = {
  id: 1,
  name: ``,
  email: ``,
  avatarUrl: ``
};
