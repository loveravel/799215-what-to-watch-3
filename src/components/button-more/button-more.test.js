import React from "react";
import renderer from "react-test-renderer";

import {FilmsCount} from "../../constants.js";
import {films} from "../../mocks/test-data.js";

import {ButtonMore} from "./button-more.jsx";

it(`<ButtonMore /> is rendered correctly with a small number of films`, () => {
  const props = {
    filteredFilms: films,
    filmsCount: FilmsCount.INITIAL_MAIN,
    incrementFilmsCount: () => {}
  };

  const tree = renderer
    .create(<ButtonMore {...props}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`<ButtonMore /> is rendered correctly with a large number of films`, () => {
  const props = {
    filteredFilms: [...films, ...films, ...films],
    filmsCount: FilmsCount.INITIAL_MAIN,
    incrementFilmsCount: () => {}
  };

  const tree = renderer
    .create(<ButtonMore {...props}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
