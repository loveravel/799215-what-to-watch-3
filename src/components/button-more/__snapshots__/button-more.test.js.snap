// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonMore /> is rendered correctly with a large number of films 1`] = `
<div
  className="catalog__more"
>
  <button
    className="catalog__button"
    onClick={[Function]}
    type="button"
  >
    Show more
  </button>
</div>
`;

exports[`<ButtonMore /> is rendered correctly with a small number of films 1`] = `null`;
